web/src/components/network/BindingSettingsForm.tsx

```diff
@@ -0,0 +1,246 @@
+  const devices = useNetworkDevices();
+
+  const labelAttributes =
+    valueKey === "macAddress" ? ["macAddress", "name"] : ["name", "macAddress"];
+
+  return (
+    <FormSelect value={value} {...formSelectProps}>
+      {devices.map((device, index) => (
+        <FormSelectOption
+          key={index}
+          value={device[valueKey]}
+          label={labelAttributes.map((attr) => device[attr]).join(" - ")}
+        />
+      ))}
+    </FormSelect>
+  );
+}
+
+/**
+ * Represents the form state.
+ */
+type FormState = {
+  /**
+   * The  binding mode for the connection
+   *  - "none":  No specific interface binding.
+   *  - "mac":   Bind to a specific MAC address.
+   *  - "iface": Bind to a specific interface name.
+   */
+  mode: "none" | "mac" | "iface";
+  /** The selected interface name for "iface" mode */
+  iface: Device["name"];
+  /** The selected MAC address for "mac" mode */
+  mac: Device["macAddress"];
+};
+
+/**
+ * Supported form actions.
+ */
+type FormAction =
+  | { type: "SET_MODE"; mode: FormState["mode"] }
+  | { type: "SET_IFACE"; iface: FormState["iface"] }
+  | { type: "SET_MAC"; mac: FormState["mac"] };
+
+/**
+ * Reducer for form state updates.
+ */
+const formReducer = (state: FormState, action: FormAction): FormState => {
+  switch (action.type) {
+    case "SET_MODE": {
+      return { ...state, mode: action.mode };
+    }
+
+    case "SET_IFACE": {
+      return { ...state, iface: action.iface };
+    }
+
+    case "SET_MAC": {
+      return { ...state, mac: action.mac };
+    }
+  }
+};
+
+/**
+ * Calculates the initial binding mode for the form based on connection data.
+ */
+const initialMode = (connection: Connection): FormState["mode"] => {
+  if (connection.macAddress) {
+    return "mac";
+  } else if (connection.iface) {
+    return "iface";
+  } else {
+    return "none";
+  }
+};
+
+/**
+ * Allows to configure how a network connection is associated with a specific
+ * network interface.
+ *
+ * Users can choose to bind by interface name, MAC address, or allow the
+ * connection on any interface.
+ */
+export default function BindingSettingsForm() {
+  const { id } = useParams();
+  const { mutateAsync: updateConnection } = useConnectionMutation();
+  const connection = useConnection(id);
+  const devices = useNetworkDevices();
+  const navigate = useNavigate();
+  const [state, dispatch] = useReducer(formReducer, {
+    mode: initialMode(connection),
+    mac: connection.macAddress || devices[0].macAddress,
+    iface: connection.iface || devices[0].name,
+  });
+
+  const onSubmitForm = (e) => {
+    e.preventDefault();
+
+    const { id, ...connectionOptions } = connection;
+
+    const updatedConnection = new Connection(id, {
+      ...connectionOptions,
+      iface: state.mode === "iface" ? state.iface : undefined,
+      macAddress: state.mode === "mac" ? state.mac : undefined,
+    });
+
+    updateConnection(updatedConnection)
+      .then(() => navigate(-1))
```

Comment:
this `-1` looks quite magic. At least document why and what it do.

Reply:
It's not magic. It is just going back, documented at React Router library.

Reply:
I mean, it is navigating one step back in the history stack. It is mentioned in React Router documentation https://reactrouter.com/6.30.1/hooks/use-navigate#usenavigate and also part of the History API documentation https://developer.mozilla.org/en-US/docs/Web/API/History_API

I see no reason for adding a comment for such an expected behavior each time it is used. *

Of course, we could extract this to a custom function helper or so, but not as part of this PR.

Reply:
For better readability we should define a constant for that.

```js
const GO_BACK = -1;
...
navigate(GO_BACK);
```

Reply:
Please, no. No in this PR